java/src/jmri/jmris/srcp/JmriSRCPThrottleServer.java

```diff
@@ -106,22 +107,7 @@ public void sendStatus(int bus, int address) throws IOException {
             // and now build the output string to send
             String StatusString = "100 INFO " + bus + " GL " + address + " ";
             StatusString += isForward ? "1 " : "0 ";
-            switch (speedStepMode) {
-                case DccThrottle.SpeedStepMode14:
-                    StatusString += (int) java.lang.Math.ceil(speedSetting * 14) + " " + 14;
-                    break;
-                case DccThrottle.SpeedStepMode27:
-                    StatusString += (int) java.lang.Math.ceil(speedSetting * 27) + " " + 27;
-                    break;
-                case DccThrottle.SpeedStepMode28:
-                    StatusString += (int) java.lang.Math.ceil(speedSetting * 28) + " " + 28;
-                    break;
-                case DccThrottle.SpeedStepMode128:
-                    StatusString += (int) java.lang.Math.ceil(speedSetting * 126) + " " + 126;
-                    break;
-                default:
-                    StatusString += (int) java.lang.Math.ceil(speedSetting * 100) + " " + 100;
-            }
+            StatusString += (int) java.lang.Math.ceil(speedSetting * speedStepMode.numSteps) + " " + speedStepMode.numSteps;
```

Comment:
This will actually break the interaction with other implementations of the SRCP protocol.  

The problem is not with the DCC modes, but when anyhthing other than one of the DCC modes is used, we use the default case.  Since this is a networking protocol, we can’t arbitrarily change what we send

Reply:
Fixed.